torchci/pages/minihud/[repoOwner]/[repoName]/[branch]/[[...page]].tsx

```diff
@@ -231,12 +234,161 @@ function CommitSummaryLine({
+      )}
       <CommitLinks row={row} />
     </div>
   );
 }
 
-function CommitSummary({ row }: { row: RowData }) {
+function DurationInfo({
+  jobs,
+  prevRow,
+  expandAllDurationInfo,
+}: {
+  jobs: JobData[];
+  prevRow: RowData | undefined;
+  expandAllDurationInfo: boolean;
+}) {
+  function getAggregateTestTimes(jobs: JobData[] | undefined) {
+    return _.reduce(
+      jobs,
+      (
+        prev: {
+          [key: string]: {
+            duration: number;
+            validData: boolean;
+          };
+        },
+        cur
+      ) => {
+        // filter out rocm and macos because they tend to be variable
+        if (
+          cur.name != undefined &&
+          cur.name.includes(",") &&
+          !cur.name.includes("rocm") &&
+          !cur.name.includes("macos")
+        ) {
+          let name = cur.name.substring(0, cur.name.indexOf(","));
+          if (!(name in prev)) {
+            prev[name] = { duration: 0, validData: true };
+          }
+          if (cur.conclusion != "success" || cur.durationS === undefined) {
+            prev[name].validData = false;
+          } else {
+            prev[name].duration += cur.durationS;
+          }
+        }
+        return prev;
+      },
+      {}
+    );
+  }
+
+  const prevRowJobsAggregate = _.pickBy(
+    getAggregateTestTimes(prevRow?.jobs),
+    (value) => value.validData
+  );
+
+  function getDurationInfo(name: string, duration: number, validData: boolean) {
+    const durationString = validData ? durationHuman(duration) : "N/A";
+    var color = "black";
+    if (
+      !validData ||
+      prevRow === undefined ||
+      prevRowJobsAggregate[name] === undefined
+    ) {
+      return {
+        concerningChange: false,
+        name,
+        color,
+        duration: durationString,
+        percentChangeString: "",
+        absoluteChangeString: "",
+      };
+    }
+    const prevDuration = prevRowJobsAggregate[name].duration;
+
+    const percentChange = duration / prevDuration;
+    const percentChangeString =
+      percentChange >= 1
+        ? `+ ${((percentChange - 1) * 100).toFixed(2)}%`
+        : `- ${((1 - percentChange) * 100).toFixed(2)}%`;
+    const absoluteChange = Math.round(duration - prevDuration);
+    const absoluteChangeString =
+      absoluteChange >= 0
+        ? `+ ${durationHuman(absoluteChange)}`
+        : `- ${durationHuman(Math.abs(absoluteChange))}`;
+    const concerningChange = Math.abs(absoluteChange) > 60 * 30;
+    if (concerningChange) {
+      color = absoluteChange > 0 ? "red" : "purple";
+    }
+    return {
+      concerningChange,
+      color,
+      name,
+      duration: durationString,
+      percentChangeString,
+      absoluteChangeString,
+    };
+  }
+
+  const [concerning, notConcerning] = _.partition(
+    _.map(getAggregateTestTimes(jobs), (value, key) => {
+      return getDurationInfo(key, value.duration, value.validData);
+    }),
+    (e) => e.concerningChange
+  );
+
+  function getRow(val: {
```

Comment:
If this is a functional component, we should just make it a component over calling a function that returns some JSX. 

Ex:

on line 368 you can do `<Row ...val>` vs `getRow(val)`